compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java

```diff
@@ -277,7 +277,9 @@ public void addSourceFile(TextDocument textDocument, String filenamePrefix, Modu
 
         @Override
         public void addSourceFile(TextDocument textDocument, String filenamePrefix) {
-            addSourceFile(textDocument, filenamePrefix, defaultModuleId);
+            if (this.currentPackage.project().kind().equals(ProjectKind.BUILD_PROJECT)) {
```

Comment:
Do we need to inform the user here that source files cannot be added to single file project?

Reply:
We can report a WARNING if it is a single file

Reply:
Addressed in 2db6400 to show a warning as follows:
```
Compiling source
	hello_world.bal
WARNING [:(1:1,1:1)] skipping addition of generated file dummyfunc. sourcefile generation is not supported with standalone bal files

Generating executable
	bar.jar
```